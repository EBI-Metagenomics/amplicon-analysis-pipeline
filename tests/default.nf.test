nextflow_pipeline {
    name "Test Pipeline Default"
    script "../main.nf"
    tag "pipeline"
    tag "default"
 

    test("-profile test") {
        when {
            params {
                
                input = "${projectDir}/tests/data/samplesheet.csv"
                outdir = "$outputDir"

                // Silva databases
                ssu_db_fasta = "${projectDir}/tests/data/SILVA/mini_SILVA-SSU.fasta"
                ssu_db_tax = "${projectDir}/tests/data/SILVA/mini_SILVA-SSU-tax.txt"
                ssu_db_otu = "${projectDir}/tests/data/SILVA/mini_SILVA-SSU.otu"
                ssu_db_mscluster = "${projectDir}/tests/data/SILVA/mini_SILVA-SSU.fasta.mscluster"
                ssu_label = "SSU"

                // UNITE database
                unite_db_fasta = "${projectDir}/tests/data/UNITE/mini_UNITE.fasta"
                unite_db_tax = "${projectDir}/tests/data/UNITE/mini_UNITE-tax.txt"
                unite_db_otu = "${projectDir}/tests/data/UNITE/mini_UNITE.otu"
                unite_db_mscluster = "${projectDir}/tests/data/UNITE/mini_UNITE.fasta.mscluster"
                unite_label = "UNITE"

                // ITSone database
                itsone_db_fasta = "${projectDir}/tests/data/ITSone/mini_ITSone.fasta"
                itsone_db_tax = "${projectDir}/tests/data/ITSone/mini_ITSone-tax.txt"
                itsone_db_otu = "${projectDir}/tests/data/ITSone/mini_ITSone.otu"
                itsone_db_mscluster = "${projectDir}/tests/data/ITSone/mini_ITSone.fasta.mscluster"
                itsone_label = "ITSonedb"

                // PR2 database
                pr2_db_fasta = "${projectDir}/tests/data/PR2/mini_PR2.fasta"
                pr2_db_tax = "${projectDir}/tests/data/PR2/mini_PR2-tax.txt"
                pr2_db_otu = "${projectDir}/tests/data/PR2/mini_PR2.otu"
                pr2_db_mscluster = "${projectDir}/tests/data/PR2/mini_PR2.fasta.mscluster"
                pr2_label = "PR2"

                dada2_pr2_label = "DADA2-PR2"

                // rfam
                rrnas_rfam_covariance_model = "${projectDir}/tests/data/rfam/mini_ribo_RF00177.cm"
                rrnas_rfam_claninfo = "${projectDir}/tests/data/rfam/mini_ribo.clan_info"

            }
        }
        then {
            // stable_name: All files + folders in ${params.outdir}/ with stable names
            def stable_name = getAllFilesFromDir(
                params.outdir,
                relative: true,
                includeDir: true,
                ignore: ['pipeline_info/*.{html,json,txt}', 'pipeline_info/execution_*.{html,txt}']
            )
 
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(
                params.outdir,
                ignoreFile: 'tests/.nftignore'
            )
 
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable file/folder names (relative paths)
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
