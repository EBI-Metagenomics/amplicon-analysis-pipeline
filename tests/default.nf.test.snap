{
    "-profile test": {
        "content": [
            61,
            [
                "SRR9674618",
                "SRR9674618/amplified-region-inference",
                "SRR9674618/amplified-region-inference/SRR9674618.16S.V3.txt",
                "SRR9674618/amplified-region-inference/SRR9674618.tsv",
                "SRR9674618/asv",
                "SRR9674618/asv/16S-V3",
                "SRR9674618/asv/16S-V3/SRR9674618_16S-V3_asv_read_counts.tsv",
                "SRR9674618/asv/SRR9674618_DADA2-PR2_asv_tax.tsv",
                "SRR9674618/asv/SRR9674618_DADA2-SILVA_asv_tax.tsv",
                "SRR9674618/asv/SRR9674618_asv_seqs.fasta",
                "SRR9674618/asv/SRR9674618_dada2_stats.tsv",
                "SRR9674618/primer-identification",
                "SRR9674618/primer-identification/SRR9674618.cutadapt.json",
                "SRR9674618/primer-identification/SRR9674618_primer_validation.tsv",
                "SRR9674618/primer-identification/SRR9674618_primers.fasta",
                "SRR9674618/qc",
                "SRR9674618/qc/SRR9674618.fastp.fastq.gz",
                "SRR9674618/qc/SRR9674618.fastp.json",
                "SRR9674618/qc/SRR9674618_multiqc_report.html",
                "SRR9674618/qc/SRR9674618_seqfu.tsv",
                "SRR9674618/qc/SRR9674618_suffix_header_err.json",
                "SRR9674618/sequence-categorisation",
                "SRR9674618/sequence-categorisation/SRR9674618.tblout.deoverlapped",
                "SRR9674618/sequence-categorisation/SRR9674618_ITS_rRNA.fa",
                "SRR9674618/sequence-categorisation/SRR9674618_SSU.fasta",
                "SRR9674618/sequence-categorisation/SRR9674618_SSU_rRNA_bacteria.RF00177.fasta",
                "SRR9674618/taxonomy-summary",
                "SRR9674618/taxonomy-summary/DADA2-PR2",
                "SRR9674618/taxonomy-summary/DADA2-PR2/SRR9674618_16S-V3.html",
                "SRR9674618/taxonomy-summary/DADA2-PR2/SRR9674618_16S-V3_DADA2-PR2_asv_krona_counts.txt",
                "SRR9674618/taxonomy-summary/DADA2-PR2/SRR9674618_DADA2-PR2.mseq",
                "SRR9674618/taxonomy-summary/DADA2-SILVA",
                "SRR9674618/taxonomy-summary/DADA2-SILVA/SRR9674618_16S-V3.html",
                "SRR9674618/taxonomy-summary/DADA2-SILVA/SRR9674618_16S-V3_DADA2-SILVA_asv_krona_counts.txt",
                "SRR9674618/taxonomy-summary/DADA2-SILVA/SRR9674618_DADA2-SILVA.mseq",
                "SRR9674618/taxonomy-summary/ITSonedb",
                "SRR9674618/taxonomy-summary/ITSonedb/SRR9674618.html",
                "SRR9674618/taxonomy-summary/ITSonedb/SRR9674618_ITSonedb.mseq",
                "SRR9674618/taxonomy-summary/ITSonedb/SRR9674618_ITSonedb.tsv",
                "SRR9674618/taxonomy-summary/ITSonedb/SRR9674618_ITSonedb.txt",
                "SRR9674618/taxonomy-summary/PR2",
                "SRR9674618/taxonomy-summary/PR2/SRR9674618.html",
                "SRR9674618/taxonomy-summary/PR2/SRR9674618_PR2.mseq",
                "SRR9674618/taxonomy-summary/PR2/SRR9674618_PR2.tsv",
                "SRR9674618/taxonomy-summary/PR2/SRR9674618_PR2.txt",
                "SRR9674618/taxonomy-summary/SSU",
                "SRR9674618/taxonomy-summary/SSU/SRR9674618.html",
                "SRR9674618/taxonomy-summary/SSU/SRR9674618_SSU.mseq",
                "SRR9674618/taxonomy-summary/SSU/SRR9674618_SSU.tsv",
                "SRR9674618/taxonomy-summary/SSU/SRR9674618_SSU.txt",
                "SRR9674618/taxonomy-summary/UNITE",
                "SRR9674618/taxonomy-summary/UNITE/SRR9674618.html",
                "SRR9674618/taxonomy-summary/UNITE/SRR9674618_UNITE.mseq",
                "SRR9674618/taxonomy-summary/UNITE/SRR9674618_UNITE.tsv",
                "SRR9674618/taxonomy-summary/UNITE/SRR9674618_UNITE.txt",
                "bco.json",
                "pipeline_info",
                "pipeline_info/software_versions.yml",
                "qc_passed_runs.csv",
                "study_multiqc_report.html"
            ],
            [
                "SRR9674618.16S.V3.txt:md5,bcb6305e3b2ee3dd9642fe5dfd18e4f7",
                "SRR9674618.tsv:md5,ee5680fce435967065a80cdf4af4a9e4",
                "SRR9674618_16S-V3_asv_read_counts.tsv:md5,cdc699a30e23459e4253e06daa08f589",
                "SRR9674618_DADA2-PR2_asv_tax.tsv:md5,03da9307218321d26d253cab713ef92e",
                "SRR9674618_DADA2-SILVA_asv_tax.tsv:md5,c03c98eb6f07ad7997bf87c7aa7fd752",
                "SRR9674618_asv_seqs.fasta:md5,42f79bbeef2e2884ef5b5a6ebca2b2fb",
                "SRR9674618_dada2_stats.tsv:md5,0e85b63d6db408972afdf6bdac53bee9",
                "SRR9674618.cutadapt.json:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SRR9674618_primer_validation.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SRR9674618_primers.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SRR9674618.fastp.fastq.gz:md5,34b5753f73925cdab80019032f8dafb4",
                "SRR9674618.fastp.json:md5,9698e9d41c4349600b01b7a376bbd84c",
                "SRR9674618_seqfu.tsv:md5,71f9c1f7f040331c45bf107f250a834a",
                "SRR9674618_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SRR9674618.tblout.deoverlapped:md5,8c44778df333a38f7a010d2d3681866a",
                "SRR9674618_ITS_rRNA.fa:md5,8ac20aebec78e89d540e7ded7511438e",
                "SRR9674618_SSU.fasta:md5,58b5d1f9f6317950779e0a0fcd7450a5",
                "SRR9674618_SSU_rRNA_bacteria.RF00177.fasta:md5,58b5d1f9f6317950779e0a0fcd7450a5",
                "SRR9674618_16S-V3_DADA2-PR2_asv_krona_counts.txt:md5,17fd4c01695b9b4026b713bcb4608947",
                "SRR9674618_DADA2-PR2.mseq:md5,ee9f1b14d00399483ce58bbbd814d010",
                "SRR9674618_16S-V3_DADA2-SILVA_asv_krona_counts.txt:md5,99840f58d62a5b42535aabc834029572",
                "SRR9674618_DADA2-SILVA.mseq:md5,ff29b0df3341ecb1b499cf241a80b8a1",
                "SRR9674618.html:md5,1766cd2dbe1a166b60a74d3b338d3901",
                "SRR9674618_ITSonedb.mseq:md5,10bea757f59b15ec74495c7610ca41c2",
                "SRR9674618_ITSonedb.tsv:md5,4a82491d9e355365f9a85111ee24218f",
                "SRR9674618_ITSonedb.txt:md5,832418ec84bc8c31b8003019c2538419",
                "SRR9674618_PR2.tsv:md5,5825974f0f87bde2636e80214085655d",
                "SRR9674618_PR2.txt:md5,6759e8a1567c0e0da9d46fc6e7086e9c",
                "SRR9674618_SSU.tsv:md5,f6bbc8100c9d0046d91c2acb5d8ced56",
                "SRR9674618_SSU.txt:md5,66ee5e8e35d54d01b38fd12493f97d83",
                "SRR9674618_UNITE.mseq:md5,0b6ec404495c6d540812efdb8163276a",
                "SRR9674618_UNITE.tsv:md5,2d4acc457081dbac5b404126f4c03dce"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-16T10:29:12.01809761"
    }
}